apb_devel/index.adoc

```diff
@@ -0,0 +1,201 @@
+[[apb-devel-intro]]
+= Introduction
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[apb-devel-intro-about]]
+== About This Guide
+
+This guide outlines the design concepts and workflow of
+link:https://github.com/ansibleplaybookbundle/ansible-playbook-bundle[Ansible Playbook Bundles (APBs)], shows how to install and use the `apb` CLI tooling,
+and provides a tutorial and reference material on writing your own APBs.
+
+[[apb-devel-intro-design]]
+== Design Overview
+
+An APB is a lightweight application definition that borrows several concepts
+from the link:https://github.com/projectatomic/nulecule[Nulecule] and
+link:http://www.projectatomic.io/docs/atomicapp/[Atomicapp] projects, namely the
+concept of a short-lived container with the sole purpose of orchestrating the
+deployment of the intended application. For the case of APBs, this short-lived
+container is the APB itself: a container with an
+link:https://www.ansible.com/[Ansible] runtime environment plus any files
+required to assist in orchestration, such as playbooks, roles, and extra
+dependencies.
+
+The
+xref:../architecture/service_catalog/ansible_service_broker.adoc#arch-ansible-service-broker[OpenShift Ansible broker] (OAB) is an implementation of the Open Service Broker (OSB) API
+that manages applications defined by APBs. The OAB is supported and deployed by
+default starting in {product-title} 3.7.
+
+Specification of an APB is intended to be lightweight, consisting of several
+named playbooks and a metadata file to capture information such as parameters to
+pass into the application.
+
+[[apb-devel-intro-workflow]]
+== Workflow
+
+The APB workflow is broken up into the following steps:
+
+. Preparation
+.. APB initialization
+.. APB spec file
+.. Actions (provision, deprovision, bind, unbind)
+. Build
+. Deploy
+
+[[apb-devel-intro-prep]]
+=== Preparation
+
+[[apb-devel-intro-apb-init]]
+==== APB Initialization
+
+////
+image::apb-prepare.png[]
+////
+
+The first step to creating an APB is to run the `apb init` command, which
+creates the required skeleton directory structure and a few required files (for
+example, the *_apb.yml_* spec file) for the APB.
+
+The following shows an example directory structure of an APB:
+
+[[apb-directory-structure]]
+.Directory Structure
+----
+example-apb/
+├── Dockerfile
+├── apb.yml
+└── roles/
+│   └── example-apb-openshift
+│       ├── defaults
+│       │   └── main.yml
+│       └── tasks
+│           └── main.yml
+└── playbooks/
+    └── provision.yml
+    └── deprovision.yml
+    └── bind.yml
+    └── unbind.yml
+----
+
+[[apb-devel-intro-spec-file]]
+==== APB Spec File
+
+An APB spec file (*_apb.yml_*) must be edited for your specific application. For
+example, the `etherpad-apb` spec file looks as follows:
+
+----
+name: fusor/etherpad-apb
```

Comment:
this should just be `name: etherpad-apb`. We remove the organization from the spec.

Reply:
The entire spec declaration is old, I would copy from here: https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/blob/master/docs/developers.md#apb-spec-file